### PR TITLE
[FIX] Added capability to pass retention at webhook endpoint level

### PR DESCRIPTION
**Pull request by nonibansal**

### CODE REVIEW

**Comment by aditya-nambiar** [comment]
Also lets not increase the version, i'l do that once by CDC upsert changes are landed in the server

**Review by nikhilgarg28 — COMMENTED** [review]
Nit - I think retention should be specified at the webhook level, not at the endpoint level. Otherwise the isomorphism checks become trickier.

File: fennel/connectors/connectors.py [hunk 1]
```diff
@@ -20,6 +23,7 @@
 SINK_FIELD = "__fennel_data_sinks__"
 DEFAULT_EVERY = Duration("30m")
 DEFAULT_DISORDER = Duration("14d")
+DEFAULT_WEBHOOK_RETENTION = Duration("14d")
```
[the remark this answers is not in the document]
Comment (nonibansal) [reply fennel/connectors/connectors.py:26]: Will make it 7 days

**Comment by nikhilgarg28** [comment]
> Nit - I think retention should be specified at the webhook level, not at the endpoint level. Otherwise the isomorphism checks become trickier.

@nonibansal / @aditya-nambiar any thoughts on this?

**Comment by nonibansal** [comment]
> Nit - I think retention should be specified at the webhook level, not at the endpoint level. Otherwise the isomorphism checks become trickier.

Me and Aditya discussed to make it endpoint ie (topic wise). Should I change the implementation?

**Comment by nonibansal** [comment]
> > Nit - I think retention should be specified at the webhook level, not at the endpoint level. Otherwise the isomorphism checks become trickier.
> 
> @nonibansal / @aditya-nambiar any thoughts on this?

@nikhilgarg28 I was thinking endpoint as equivalent of a topic, so wanted to have to retention of the topic defined at endpoint level ie at dataset level, not at group of datasets.

**Comment by aditya-nambiar** [comment]
@nonibansal  lets keep it at the webhook level. Currently all datasets that share the webhook, share the topic underneath it. If different datasets use the same webhook but with different retention, it will be harder to handle. Its a corner case, but might just be easier to keep the retention at the webhook level.

**Comment by nikhilgarg28** [comment]
I'm okay with that too.

**Review by aditya-nambiar — APPROVED** [review]

**Pull request merged**